umbraco/Umbraco-CMS
# Omit entirely un-routable content from Delivery API output

### Prerequisites


### Description

In a culture variant setup with no hostname mappings assigned at root level, Umbraco is only able to route the default culture, unless `HideTopLevelNodeFromPath` is set to `false` (default is `true`).

With the introduction of the `Start-Item` header, the Delivery API is able to route all cultures in the same scenario. Unfortunately, this routing of otherwise un-routeable content is a little too lenient in certain scenarios, allowing for outputting descendants in a culture that is not published all the way up the content tree.

The result is content output with invalid `path` values:

<img width="357" alt="image" src="https://github.com/umbraco/Umbraco-CMS/assets/7405322/4cb04e4b-dc26-42af-90f9-b43f48435f21">
<img width="292" alt="image" src="https://github.com/umbraco/Umbraco-CMS/assets/7405322/bacaabbf-388f-4f51-b1a0-4038a5b11a28">

This PR ensures that the Delivery API only outputs content that has an "unbroken chain" of published content in any given culture - that is, all ancestors of the content is either published in that culture, or is culture invariant.

### Testing this PR

1. Create a structure of content using:
   - A culture _invariant_ root with no hostname mappings applied.
   - Culture _variant_ descendants in at least two levels.
2. Publish everything in all cultures.
3. Verify that content at level two (or below) can be fetched by path and ID in all cultures using the Delivery API.
4. Unpublish content at level one in the non-default culture.
5. Verify that content at level two (or below) can no longer be fetched in the non-defaullt culture using the Delivery API, neither by path or by ID.
6. Verify that the same content can be fetched in the default culture, and that the `cultures` collection of the output does not contain the non-default culture.
